### ISSUE TITLE
Get instance status has high frequency `SHOW MASTER STATUS`, which could block MySQL written data.

### ISSUE THREAD

**Issue by ywhywl** [openark/orchestrator#1331]
If MySQL is committing a large transaction ,purge a large number of binlog files or  Other scenarios,  it may take 5S or longer .

During this time, the `SHOW MASTER STATUS` will be blocked, and the blocked `SHOW MASTER STAUS` will block all write requests, which may cause the commit to timeout

I find that the information obtained by `SHOW MASTER STATUS` is mainly used for web display. 
Can we  Add a  switch  for `SHOW MASTER STATUS` to 'open or close'?

**Comment by shlomi-noach** [comment]
> I find that the information obtained by SHOW MASTER STATUS is mainly used for web display.

The information is used to determine the binlog coordinates of an instance, used in:

- graceful takeover
- pseudo-GTID operations

If we were to disable reading `SHOW MASTER STATUS` on a "normal" probe, then:

- we would not be able to run pseudo-GTID-based recoveries (which is fine for people who use GTID)
- we'd need a way to tell `orchestrator` to read `SHOW MASTER STATUS` when running specific operations like graceful takeover.

**Comment by ywhywl** [comment]
I agree with you, we'd need a way to tell `orchestrator `to read` SHOW MASTER STATUS` when running specific operations like graceful takeover

**some times, we do not accept that HA components affect database availability ,such as large trading scenario**

If gtid_mode=on, FAILOVER don't need imformation of` SHOW MASTER STATUS` ,We can give the user a choice,Turn on or off `SHOW MASTER STATUS` switch, is greateful:

- This switch default "on"
- if on: `Orchestrator `work as it do now 
- if off：`Orchestrator` not read `SHOW MASTER STATUS `normally, but when we need it, such as befor graceful takeover or other scenarios

**Comment by yzsfly** [comment]
we recently met the same issue. So any update or workaround on this ?
